omnipaxos/src/omni_paxos.rs

```diff
@@ -62,6 +67,31 @@ impl OmniPaxosConfig {
             !self.peers.contains(&self.pid),
             "Peers should not include self pid"
         );
+
+        let num_nodes = self.peers.len() + 1;
+        match (self.leader_quorum_size, self.append_quorum_size) {
+            (Some(prepare_quorum_size), Some(accept_quorum_size)) => {
+                let quorums_overlap = prepare_quorum_size + accept_quorum_size > num_nodes;
+                assert!(
+                    quorums_overlap,
+                    "The sum of quorum sizes must exceed the # of nodes"
+                );
+                assert!(
+                    prepare_quorum_size > 0 && prepare_quorum_size <= num_nodes,
+                    "Prepare quorum must be in range 0 to # of nodes in the cluster"
+                );
+                // TODO: currently accept_quorum_size = 1 doesn't work, add to check?
+                assert!(
+                    accept_quorum_size > 0 && accept_quorum_size <= num_nodes,
+                    "Accept quorum must be in range 0 to # of nodes in the cluster"
+                );
+                assert!(prepare_quorum_size >= accept_quorum_size, "Leader election cannot guarantee progress if prepare quorum is smaller than accept quorum");
+            }
+            (None, Some(_)) => panic!("Prepare quorum must also be defined"),
+            (Some(_), None) => panic!("Accept quorum must also be defined"),
+            (None, None) => (),
```

Comment:
To make the code clean both here and in Sequence Paxos when checking quorums we should have an enum:

Quorum {
      Majority,
      Flexible(usize, usize)
}

where the default is Majority. Then we can also have convenient utility functions:

impl QuorumConfig {
     fn got_accept_quorum -> bool {}
     fn got_prepare_quorum -> bool {}
}

Reply:
Maybe we have the struct without the functions and once it gets passed to `SequencePaxos::build` we go back to the `prepare_quorum_size` and `accept_quorum_size` fields.

Reply:
Good point. I would say let's keep the flexible quorum field as an Option<(usize, usize)> in OmniPaxosConfig. 

In Sequence Paxos, I think we can use `Majority(usize)` to avoid recalculations. I also still prefer the shared functions to keep the code clean.